lib/scope.js

```diff
@@ -909,36 +921,73 @@ module.exports = {
     // data for `users[0]`, there's no need to check the trigger var.
     // That way devs with simple cases can ignore the trigger var column.
 
-    // If field uses proxies and there has been more than one match
-    // we now use the trigger var to narrow down the matches.
-    if ( uses_proxies && matches.length > 1 ) {
-      // Multiple matches means this came from a story
-
-      // Feedback for devs - we need the trigger var but we don't have it (field data)
-      // TODO: Add docs and link to docs on how to add the trigger var
-      if ( field.trigger === undefined ) { await scope.addToReport(scope, { type: `warning`, value: `A field on this page uses an index variable or a generic object, but the page does not contain an element with the trigger variable's name. Tell the developer to add an element to the page with the id "trigger"` }); }
+    // Make sure the trigger var matches for a page that uses proxies. Only
+    // story tables need this disambiguation.
+    if ( trigger_might_be_needed && uses_proxies && matches.length > 0 ) {
+
+      if ( !field.trigger ) {
+        // field/page trigger var is needed but is missing.
+        let page_trigger_warning = `A field on this page uses an index `
+          + `variable or a generic object, but the interview page does not `
+          + `give the trigger variable's name. See `
+          + `https://suffolklitlab.org/docassemble-AssemblyLine-documentation`
+          + `/docs/automated_integrated_testing/#a-missing-trigger-variable.`;
+        await scope.addToReport(scope, { type: `warning`, value: page_trigger_warning });
+      }
 
       let original_matches = matches;
+      let fallbacks = [];
       matches = [];
       for ( let match of original_matches ) {
-        // Feedback for devs - we need the trigger var but we don't have it (var data)
-        if ( match.trigger === undefined ) { await scope.addToReport(scope, { type: `warning`, value: `A field on this page uses an index variable or a generic object. The test gave story data for setting the field, but that data is missing the trigger. Add a trigger variable to the story table \`trigger\` column. For now, the test may end up ignoring this field. The row's data is\n${ JSON.stringify( match.original )}.\nThe field's data is\n${ JSON.stringify( field )}` }); }
-          // TODO: Add a warning if they're both `undefined`?
-        else {
-
+        if ( !match.trigger ) {
```

Comment:
Same as above: we have decided to consider calling this match.triggerName.

Reply:
This may become a future issue to do properly because it goes deep. We may be able to do something more superficial in here, though.